### PR TITLE
Booking date validation and autorizations

### PR DESCRIPTION
**Pull request by SVILLEBRUN**

### CODE REVIEW

File: app/views/jewels/show.html.erb [hunk 1]
```diff
@@ -6,3 +6,6 @@
   <%= cl_image_tag photo.key, height: 300, width: 400, crop: :fill %>
 <% end %>
 <%= link_to "All our jewels", jewels_path %>
+
+<%= link_to "Edit this jewle", edit_jewel_path(@jewel) if policy(@jewel).edit?%>
```
Comment (XxSachaxX) [inline app/views/jewels/show.html.erb]: "Edit this "Jewel"

File: app/controllers/jewels_controller.rb [hunk 2]
```diff
@@ -25,17 +26,17 @@ def create
...
 
   def edit
-    @jewel = Jewel.find(params[:id])
+    authorize @jewel
   end
 
   def update
-    @jewel = Jewel.find(params[:id])
+    authorize @jewel
     @jewel.update(jewels_params)
```
Comment (XxSachaxX) [inline app/controllers/jewels_controller.rb]: Probablement ajouter:
if @jewel.update(jewel_params) 
redirect_to jewel_path(@jewel), notice: "Jewel updated"
else
render :new status: :unprocessable entity

**Review by XxSachaxX — CHANGES_REQUESTED** [review]

File: app/views/jewels/show.html.erb (hunk 1 above)
Comment (SVILLEBRUN) [reply app/views/jewels/show.html.erb]: Thks

File: app/controllers/jewels_controller.rb (hunk 2 above)
Comment (SVILLEBRUN) [reply app/controllers/jewels_controller.rb]: done

**Pull request merged**